export/securedrop_export/disk/service.py

```diff
@@ -1,114 +1,167 @@
 import logging
 
-from securedrop_export.archive import Archive
-
 from .cli import CLI
 from .status import Status
-from .volume import Volume, MountedVolume
+from .volume import Volume, MountedVolume, EncryptionScheme
+from securedrop_export.archive import Archive
 from securedrop_export.exceptions import ExportException
+from typing import List, Optional, Tuple
 
 
 logger = logging.getLogger(__name__)
 
 
 class Service:
     """
-    Checks that can be performed against the device(s).
+    Actions that can be performed against USB device(s).
     This is the "API" portion of the export workflow.
     """
 
-    def __init__(self, cli: CLI):
+    def __init__(self, submission: Archive, cli: CLI = CLI()):
         self.cli = cli
+        self.submission = submission
 
     def scan_all_devices(self) -> Status:
         """
         Check all connected devices and return current device
         status.
         """
         try:
-            all_devices = self.cli.get_connected_devices()
-            number_devices = len(all_devices)
-
-            if number_devices == 0:
-                return Status.NO_DEVICE_DETECTED
-            elif number_devices > 1:
-                return Status.MULTI_DEVICE_DETECTED
+            volumes = self.cli.get_all_volumes()
+            if len(volumes) == 0:
+                status = Status.NO_DEVICE_DETECTED
+            elif len(volumes) > 1:
+                status = Status.MULTI_DEVICE_DETECTED
             else:
-                return self.scan_single_device(all_devices[0])
+                status, _ = self._check_volumes(volumes)
```

Comment:
On first reading, I thought there was something to DRY here.  There isn't, but it may be worth making that explicit:

```suggestion
                # NB. Here we've checked for exactly 1 device.  _check_volumes() will also check for exactly 1 volume.
                status, _ = self._check_volumes(volumes)
```